### PR TITLE
[Wav2vec2 + LM Test] Improve wav2vec2 with lm tests and make torch version dependent for now

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Trying to correct potentially flaky test: 
```
tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py::Wav2Vec2ProcessorWithLMTest::test_word_time_stamp_integration
```
The test actually always passed for me locally.

I've now changed list to tensor as list comparison with float numbers seems brittle. @ydshieh - just to better understand, is this test consistently failing or was it flaky?

### CODE REVIEW

**Comment by ydshieh** [comment]
It fails consistently since PT 1.12.

**Comment by patrickvonplaten** [comment]
Note after some more debugging the reason seems to be in `datasets`. E.g. this part of the test:

```python
ds = load_dataset("common_voice", "en", split="train", streaming=True)
ds = ds.cast_column("audio", datasets.Audio(sampling_rate=16_000))
ds_iter = iter(ds)
sample = next(ds_iter)
print(sample["audio"]["array"])
```

yields different results between `torchaudio=1.11` and `torchaudio=1.12`. Opening an issue in Datasets.

File: tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py [hunk 1]
```diff
@@ -411,7 +412,9 @@ def test_word_time_stamp_integration(self):
         model = Wav2Vec2ForCTC.from_pretrained("patrickvonplaten/wav2vec2-base-100h-with-lm")
 
         # compare to filename `common_voice_en_100038.mp3` of dataset viewer on https://huggingface.co/datasets/common_voice/viewer/en/train
+        print("Out", np.sum(np.abs(sample["audio"]["array"])))
         input_values = processor(sample["audio"]["array"], return_tensors="pt").input_values
+        print("Out PT", input_values.abs().sum())
```
Comment (patrickvonplaten) [inline tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py:417]:
```suggestion
```

File: tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py [hunk 2]
```diff
@@ -411,7 +412,9 @@ def test_word_time_stamp_integration(self):
         model = Wav2Vec2ForCTC.from_pretrained("patrickvonplaten/wav2vec2-base-100h-with-lm")
 
         # compare to filename `common_voice_en_100038.mp3` of dataset viewer on https://huggingface.co/datasets/common_voice/viewer/en/train
+        print("Out", np.sum(np.abs(sample["audio"]["array"])))
```
Comment (patrickvonplaten) [inline tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py:415]:
```suggestion
```

**Comment by patrickvonplaten** [comment]
@ydshieh good for a second review

File: tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py [hunk 3]
```diff
@@ -435,21 +436,19 @@ def test_word_time_stamp_integration(self):
...
-            end_times,
-            [
-                1.54, 1.88, 2.14, 2.46, 2.9, 3.18, 3.54, 3.72, 4.02, 4.18, 4.76, 5.16, 5.56, 5.7, 5.86, 6.2, 6.38, 6.62, 6.94,
-            ],
-        )
+        expected_start_tensor = torch.tensor([1.42, 1.64, 2.12, 2.26, 2.54, 3.0, 3.24, 3.6, 3.8, 4.1, 4.26, 4.94, 5.28, 5.66, 5.78, 5.94, 6.32, 6.54, 6.66])
+
+        # TODO(Patrick): This if-else version statement should be removed once
+        # https://github.com/huggingface/datasets/issues/4889 is resolved
+        if version.parse(torch.__version__) >= version.parse("1.12.0"):
```
Comment (ydshieh) [inline tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py:447]: Maybe let's use what has been done in
https://github.com/huggingface/transformers/pull/18460
?

**Review by ydshieh — APPROVED** [review]
Thank you @patrickvonplaten . LGTM, but probably a small change to make it better.

File: tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py (hunk 3 above)
Comment (patrickvonplaten) [reply tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py:447]: Yes definitely - thanks!

Comment (patrickvonplaten) [inline tests/models/wav2vec2_with_lm/test_processor_wav2vec2_with_lm.py:447]:
```suggestion
        if version.parse(version.parse(torch.__version__).base_version >= version.parse("1.12.0"):
```

**Pull request merged**